### PR TITLE
19496 - Refactor EFT Short names (multiple account links)

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/19496

*Description of changes:*
- update models to support multiple account linking and states
- migrate existing data to new eft_short_name_links table
- clean up old short name patch routes (this is replaced by creating links on a short name)
- implemented initial get links route - more details will be filled out in subsequent tickets (EFT Payment details screen)
- update existing logic to support the new linking table

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/models/eft_credit.py [hunk 1]
```diff
@@ -61,11 +61,3 @@ class EFTCredit(BaseModel):  # pylint:disable=too-many-instance-attributes
     def find_by_payment_account_id(cls, payment_account_id: int):
         """Find EFT Credit by payment account id."""
         return cls.query.filter_by(payment_account_id=payment_account_id).all()
-
-    @classmethod
-    def update_account_by_short_name_id(cls, short_name_id: int, payment_account_id: int):
```
Comment (ochiu) [inline pay-api/src/pay_api/models/eft_credit.py:66]: Not needed anymore, as short name credits can apply to multiple accounts now via new linking table

File: pay-api/src/pay_api/models/eft_short_names.py [hunk 2]
```diff
@@ -78,9 +71,9 @@ class EFTShortnameSchema:  # pylint: disable=too-few-public-methods
     transaction_date: datetime
     deposit_date: datetime
     deposit_amount: Decimal
-    linked_by: str
-    linked_by_name: str
-    linked_on: datetime
+    updated_by: str
```
Comment (ochiu) [inline pay-api/src/pay_api/models/eft_short_names.py:74]: Renamed to be more generic due to the different statuses possible now. The status will indicate the action,

LINKED = linked by, etc

link history table will track any transitions and who did it

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 3]
```diff
@@ -53,180 +55,63 @@ class EFTShortnamesSearch:  # pylint: disable=too-many-instance-attributes
...
-    state: Optional[str] = None
+    state: Optional[List[str]] = None
     page: Optional[int] = 1
     limit: Optional[int] = 10
 
 
 class EFTShortnames:  # pylint: disable=too-many-instance-attributes
     """Service to manage EFT short name model operations."""
 
-    def __init__(self):
```
Comment (ochiu) [inline pay-api/src/pay_api/services/eft_short_names.py:64]: Removed the dao wrapper so things are a little cleaner

**Review by seeker25 — APPROVED** [review]
Great work, a couple of unit tests still busted

**Comment by ochiu** [comment]
> Great work, a couple of unit tests still busted

Merging, will update payment-job and pay-queue tests as it needs this PRs model changes

**Pull request merged**